### PR TITLE
Cleanup duplicated classes in DrawableRoomPlaylistItem

### PR DESCRIPTION
**Pull request by smoogipoo**
Also fixes missing size specification on the show results button, causing it to never show.

image

### CODE REVIEW

File: osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs [hunk 1]
```diff
@@ -59,6 +59,8 @@ public class DrawableRoomPlaylistItem : OsuRearrangeableListItem<PlaylistItem>
 
         public readonly PlaylistItem Item;
 
+        public Drawable RemoveButton { get; private set; }
```
Comment (peppy) [inline osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs:62]: Is this only exposed for testing purposes? Not sure about that..

Comment (smoogipoo) [reply osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs:62]: It is, any suggestions? I've just taken the easiest path. I don't want to `.ChildrenOfType<Button>.ElementAt(x)`.

Comment (peppy) [reply osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs:62]: Check the icon i guess? and make a local helper property to grab it in the test class.

Comment (smoogipoo) [reply osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs:62]: There's like... 4 test classes doing this. But okay.

Comment (smoogipoo) [reply osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs]: I can't do it, it's too flimsy. I've added back the class in a minimal state. If you don't agree with this then I suggest you make the change yourself.

Comment (peppy) [reply osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs]: I think that's fine, not sure why it was removed in the first place.

Comment (smoogipoo) [reply osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs]: Because there's going to be another dummy class, and that just seems silly in comparison to exposing the buttons.

**Review by peppy — APPROVED** [review]

**Comment by peppy** [comment]
Looks like some actual failures here

**Comment by smoogipoo** [comment]
It's not related to this PR, but I'll fix it here. One of the tests is in the wrong namespace so it wasn't running in my local test session.

**Comment by smoogipoo** [comment]
Actually I'm not going to fix it in this PR because it's an involved change.

tl;dr You can add an `await Task.Delay(2000)` in the async continuation of `MultiplayerClient.SettingsChanged` to repro these (and several other) failures.

**Review by peppy — APPROVED** [review]
